tf_agents/environments/tf_py_environment.py

```diff
@@ -333,7 +336,7 @@ def _set_names_and_shapes(self, step_type, reward, discount,
     batch_shape = tf.TensorShape(batch_shape)
     if not tf.executing_eagerly():
       # Shapes are not required in eager mode.
-      reward.set_shape(batch_shape)
+      reward.set_shape(batch_shape.concatenate(self._time_step_shapes[1]))
```

Comment:
Why [1]?  And why calculate all flattened shapes if you only use one of them?

Reply:
Hi @ebrevdo. Thanks for the feedback.

I agree, it is not as clear as it should be and not necessary to cache all shapes.. I've updated to not cache shapes in __init__ and to explicitly query `self.time_step_spec().reward.shape` rather than referring to it by its index position after flattening (`[1]`).